### ISSUE TITLE
False positive for `attributes` rule when using `@MainActor`

### ISSUE THREAD

**Issue by marcomasser** [realm/SwiftLint#4599]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

It seems using `@MainActor` triggers a false positive depending on where `@MainActor` is exactly. See example below.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
/Users/marco/Desktop/test/test.swift:5:7: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports. (attributes)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version 0.50.1
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules: 
  - attributes
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  > No
* Which Xcode version are you using (check `xcodebuild -version`)?
  > Xcode 14.1
  > Build version 14B47b

```swift
// This triggers a violation:
import AppKit

@NSApplicationMain
@MainActor
final class AppDelegate: NSAppDelegate {}
```

The order of the attributes is important:
```swift
// This does NOT trigger a violation:
import AppKit

@MainActor
@NSApplicationMain
final class AppDelegate: NSAppDelegate {}
```

I also tried using `@main` instead of `@NSApplicationMain` with the same result.

Interestingly, using e.g. `@main` _and_ `@NSApplicationMain` but _without_ `@MainActor` does not trigger a warning, regardless of the order. So it seems like the problem is `@MainActor`.
